zerver/views/auth.py

```diff
@@ -832,32 +833,22 @@ def api_fetch_api_key(
         request=request, username=username, password=password, realm=realm, return_data=return_data
     )
     if return_data.get("inactive_user"):
-        return json_error(
-            _("Your account has been disabled."), data={"reason": "user disable"}, status=403
-        )
+        raise AuthenticationFailureError(_("Your account has been disabled."), "user disable")
```

Comment:
The `reason` parameter doesn't super make sense; additionally, we actually have specific classes for these that we can use, so I'm going to try migrating to them.